source/hp/refinement.cc

```diff
@@ -168,8 +188,9 @@ namespace hp
               }
           }
 
-      if (const parallel::TriangulationBase<dim, spacedim> *parallel_tria =
-            dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
+      if (const parallel::distributed::Triangulation<dim, spacedim>
+            *parallel_tria = dynamic_cast<
```

Comment:
I noticed that we do not need a the succeeding communication step for the `parallel::shared::Triangulation` class, since we will use distributed vectors in that case (similar to `step-17`).

However, I totally forgot about the new `parallel::fullydistributed::Triangulation` class. I'll revert back to the general class, and add another check whether it's a `parallel::shared::Triangulation`.